src/google/cloud/ndb/model.py

```diff
@@ -4901,20 +4901,115 @@ def __init__(self, *args, **kwargs):
         raise NotImplementedError
 
 
-def transactional(*args, **kwargs):
-    raise NotImplementedError
+def transactional(**options):
+    """A decorator to run a function automatically in a transaction.
 
+    Usage example:
 
-def transactional_async(*args, **kwargs):
-    raise NotImplementedError
+    @transactional(retries=1, read_only=False)
+    def callback(args):
+        ...
+    
+    See google.cloud.ndb.transaction above for available options.
+    """
 
+    def transactional_wrapper(wrapped):
+        @functools.wraps(wrapped)
+        def transactional_inner_wrapper(*args, **kwargs):
+            def callback():
+                wrapped(*args, **kwargs)
 
-def transactional_tasklet(*args, **kwargs):
-    raise NotImplementedError
+            return _transaction.transaction(callback, **options)
 
+        return transactional_inner_wrapper
 
-def non_transactional(*args, **kwargs):
-    raise NotImplementedError
+    return transactional_wrapper
+
+
+def transactional_async(**options):
+    """A decorator to run a function automatically in an asynchronous
+       transaction.
+    
+    Usage example:
+
+    @transactional_async(retries=1, read_only=False)
+    def callback(args):
+        ...
+    
+    See google.cloud.ndb.transaction above for available options.
+    """
+
+    def transactional_async_wrapper(wrapped):
+        @functools.wraps(wrapped)
+        def transactional_async_inner_wrapper(*args, **kwargs):
+            def callback():
+                wrapped(*args, **kwargs)
+
+            return _transaction.transaction_async(callback, **options)
+
+        return transactional_async_inner_wrapper
+
+    return transactional_async_wrapper
+
+
+def transactional_tasklet(**options):
+    """A decorator that turns a function into a tasklet and runs it
+       automatically in an asynchronous transaction.
```

Comment:
I think the style is to keep first line to a single line, then expound in first following paragraph.